hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieRealtimeTableCompactor.java

```diff
@@ -122,8 +123,13 @@
 
     // Compacting is very similar to applying updates to existing file
     HoodieCopyOnWriteTable table = new HoodieCopyOnWriteTable(config, metaClient);
+
+    Optional<HoodieDataFile> oldDataFileOpt = table.getROFileSystemView()
```

Comment:
If required, you can perform this operation inside MergeHandle and remove passing it down.

Reply:
I asked Balaji to keep it like this. I actually think we should move our IOHandles to be dumb and just to an operation based on list of files passed in..  (similar to generating fileID outside CreateHandle)

Reply:
okay, we might need to perform a clean of this code then soon